### PR TITLE
adaptive ephemerality

### PR DESCRIPTION
**Pull request by davidmarkclements**
in order to support this through the stack every layer needed an update:

* https://github.com/mafintosh/dht-rpc/pull/9
* https://github.com/hyperswarm/dht/pull/27/
* https://github.com/hyperswarm/discovery/pull/15
* https://github.com/hyperswarm/network/pull/18
* https://github.com/hyperswarm/hyperswarm/pull/51

### CODE REVIEW

**Comment by mafintosh** [comment]
👍

File: index.js [hunk 1]
```diff
@@ -144,7 +144,10 @@ class Discovery extends EventEmitter {
     super()
 
     if (!opts) opts = {}
-
+    // if ephemeral is undefined, null or anything other than a boolean
+    // then this signifies adaptive ephemerality
+    opts.adaptive = typeof opts.ephemeral !== 'boolean'
```
Comment (mafintosh) [inline index.js]: can we only set the default if it's not set?

Comment (davidmarkclements) [reply index.js]: good point, fixed

**Comment by mafintosh** [comment]
1.10.0

**Pull request merged**